doc/convergence_curves.rst

```diff
@@ -0,0 +1,38 @@
+.. _convergence_curves:
+
+Sampling of convergence curves: ``stopping_strategy``
+=====================================================
+
+There are two ways to create a convergence curve for a given solver, controlled by the ``stopping_strategy`` attribute of the solver.
+
+1. Using iterations or tolerance
+--------------------------------
+
+The first way, used for black box solvers, is to create curves by calling the ``Solver.run`` method several times for each solver with different values for the ``stop_val`` parameter.
+The solver is restarted from scratch at each call to ``run``.
+- if the solver's ``stopping_strategy`` is ``"iteration"``, the number of iterations passed to ``run`` increases geometrically from 1 with a factor XXX, eventually switching to XXX if the curves becomes flat., and the
```

Comment:
Thanks for the great PR ! Could it be explained how exactly is the curve deemed flat with pointers to the lines of code that take care of that ?
I am assuming it uses heuristics similar to this dynamic learning rate scheduler (http://blog.dlib.net/2018/02/automatic-learning-rate-scheduling-that.html) ?

Reply:
@jeandut I did another pass on the PR, could you point anything that remains unclear? Thanks for the great feedback :pray:

Reply:
Done !